src/View/Helper/DatatableHelper.php

```diff
@@ -669,7 +669,12 @@ protected function processActionLink(array $link): string
         case Datatables::LINK_TYPE_POST:
             $output = new \CakeDC\Datatables\View\Formatter\Link\PostLink($this, $link);
             break;
-
+			case Datatables::LINK_TYPE_CUSTOM:
+				if (empty($link['formatter'])) {
+					throw new \OutOfBoundsException("please specify a custom formatter");
+				}
+				$output = new $link['formatter']($this,$link);
```

Comment:
I would also check the $output has a method `link` to ensure we won't fail later in line 684